src/codegen/spirv.zig

```diff
@@ -4456,16 +4480,20 @@ const DeclGen = struct {
     ) !IdRef {
         const result_ty_id = try self.resolveType(result_ptr_ty, .direct);
 
-        const mod = self.module;
-        const object_ty = object_ptr_ty.childType(mod);
-        switch (object_ty.zigTypeTag(mod)) {
-            .Struct => switch (object_ty.containerLayout(mod)) {
+        const zcu = self.module;
+        const object_ty = object_ptr_ty.childType(zcu);
+        switch (object_ty.zigTypeTag(zcu)) {
+            .Pointer => {
+                assert(object_ty.isSlice(zcu));
+                return self.accessChain(result_ty_id, object_ptr, &.{field_index});
+            },
```

Comment:
Does this mean that AIR instructions `.ptr_slice_len_ptr` and `.ptr_slice_ptr_ptr` will be deprecated? Afaict they do the exact same thing.

Reply:
No -- those instructions are for taking such pointers at *runtime*, i.e. when the slice ptr is runtime-known. It has always been permitted for constant pointers to refer to slice `len`/`ptr` in this way.